stm/stm.ml

```diff
@@ -2234,10 +2234,13 @@ let known_state ?(redefine_qed=false) ~cache id =
 	    if eff then update_global_env ()
           ), (if eff then `Yes else cache), true
       | `Cmd { cast = x; ceff = eff } -> (fun () ->
-            reach view.next;
-            stm_vernac_interp id x;
-	    if eff then update_global_env ()
-          ), (if eff then `Yes else cache), true
+          (match !Flags.async_proofs_mode with
```

Comment:
Enrico, I think this is the flag you were thinking of?

`Flags.compilation_mode` is `BuildVo | BuildVio | Vio2Vo`, none of which should apply to interactive mode.